packages/Webkul/Shop/src/Http/Controllers/API/CartController.php

```diff
@@ -220,4 +221,24 @@ public function destroyCoupon(): JsonResource
             'message'  => trans('shop::app.checkout.cart.coupon.remove'),
         ]);
     }
+
+    /**
+     * Cross-sell product listings.
+     *
+     * @return \Illuminate\Http\Resources\Json\JsonResource::collection
+     */
+    public function crossSellProducts()
+    {
+        foreach (Cart::getCart()->items as $item) {
+            $product = $this->productRepository->findOrFail($item['product_id']);
+
+            if ($product->cross_sells()->count()) {
+                $crossSellProducts = $product->cross_sells()
+                    ->take(core()->getConfigData('catalog.products.cart_view_page.no_of_cross_sells_products'))
+                    ->get();
+            }
+        }
+
+        return ProductResource::collection($crossSellProducts);
```

Comment:
Too many issues in this code, and too many queries in the loop also it will only take the last item cross cross-sell products.

Reply:
Done